truss-chains/truss_chains/model_skeleton.py

```diff
@@ -29,7 +29,8 @@ def __init__(
             definitions.Environment.model_validate(environment) if environment else None
         )
         chainlet_to_deployed_service = populate_chainlet_service_predict_urls(
-            truss_metadata.chainlet_to_service
+            truss_metadata.chainlet_to_service,
+            raise_on_missing_key=True,
```

Comment:
Why is `raise_on_missing_key` configurable? Isn't a missing entry always a fatal error from which a chain deployment cannot recover?

Reply:
Not for old Truss CLI versions, which include `predict_url`s as part of `ServiceDescriptor`s in `TrussMetadata`. So the old CLI versions will generate code that goes through the `raise_on_missing_key=False` path.